python3Packages.guestfs: 1.40.1 -> 1.40.2; mark broken on 3.13

SigmaSquadron opened the pull request:
Seems like this release refactors a bunch of deprecated python code, which in turn fixes the build.

ZHF: #403336

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

SigmaSquadron committed:
python312Packages.guestfs: 1.40.1 -> 1.40.2

Seems like this release refactors a bunch of deprecated python code,
which in turn fixes the build. Still doesn't work on 3.13 though.

Signed-off-by: Fernando Rodrigues <alpha@sigmasquadron.net>

ethancedwards8 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 406620`

Logs: https://github.com/ethancedwards8/nixpkgs-review-gha/actions/runs/14999279519

<details><summary>Download packages from cache:</summary><ul>
<li><details><summary><code>x86_64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-406620-x86_64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://ethancedwards8.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  ethancedwards8.cachix.org-1:YMasjqyFnDreRQ9GXmnPIshT3tYyFHE2lUiNhbyIxOc=
  ' \
  /nix/store/5bry4mqndwwgd98wbnqkvgzckdq32r1h-diffoscope-295 \
  /nix/store/q8xqv0v9b471m9hfz3fr1q757aywg8xi-diffoscope-295-dist \
  /nix/store/ph88fdvxfqzcvzylk5648ybpbl1dkspb-diffoscope-295-man \
  /nix/store/yqxfpvy1kpd0hk51gdkd0i3fl7cpcv8x-python3.12-guestfs-1.40.2 \
  /nix/store/iz4a2kzpp67yqxkf9cd86z510g56yrv5-python3.12-guestfs-1.40.2-dist
```
</details></li>
<li><details><summary><code>aarch64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-406620-aarch64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://ethancedwards8.cachix.org' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  ethancedwards8.cachix.org-1:YMasjqyFnDreRQ9GXmnPIshT3tYyFHE2lUiNhbyIxOc=
  ' \
  /nix/store/xqr6apql5h9isy86j0nd6c1fcmg5qab5-python3.12-guestfs-1.40.2 \
  /nix/store/5jv92cnrgckl4zf7vvihabhzbpc9i476-python3.12-guestfs-1.40.2-dist
```
</details></li>
</ul></details>


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python313Packages.guestfs</li>
    <li>python313Packages.guestfs.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>diffoscope</li>
    <li>diffoscope.dist</li>
    <li>diffoscope.man</li>
    <li>python312Packages.guestfs</li>
    <li>python312Packages.guestfs.dist</li>
  </ul>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python313Packages.guestfs</li>
    <li>python313Packages.guestfs.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python312Packages.guestfs</li>
    <li>python312Packages.guestfs.dist</li>
  </ul>
</details>

SigmaSquadron commented:
Shoot. I guess `python3Packages` resolved to 3.12 on my nixpkgs.

Do you have the logs for 3.13? I'm pretty sure we'll need to set it as broken.

ethancedwards8 commented:
https://github.com/ethancedwards8/nixpkgs-review-gha/actions/runs/14999279519/job/42141579124#step:7:10437

failed test?

SigmaSquadron committed:
python313Packages.guestfs: mark broken

Looks like it implictly calls a function that shouldn't be called implictly after 3.12.

Signed-off-by: Fernando Rodrigues <alpha@sigmasquadron.net>

SigmaSquadron commented:
Nope, looks like a compilation error with Python: `error: implicit declaration of function ‘PyEval_ThreadsInitialized’`.

`broken` it is.

FliegendeWurst commented on pkgs/development/python-modules/guestfs/default.nix:
```diff
@@ -31,5 +32,6 @@ buildPythonPackage rec {
     license = licenses.lgpl2Plus;
     maintainers = with maintainers; [ grahamc ];
     inherit (libguestfs.meta) platforms;
+    broken = pythonOlder "3.12"; # FIXME: "error: implicit declaration of function 'PyEval_ThreadsInitialized'" on Python 3.13.
```
This is fixed in 1.42 https://bugzilla.redhat.com/show_bug.cgi?id=2343777

FliegendeWurst replied:
Though that version (and newer versions) are only available on [Github](https://github.com/libguestfs/libguestfs) and require a more complicated build process.

SigmaSquadron replied:
That's libguestfs, not guestfs python.

FliegendeWurst replied:
It is the same library. They provide bindings to several languages.

SigmaSquadron replied:
oh. Still, I have no idea how to build it :p 
@grahamc can update it if it matters that much.

FliegendeWurst commented on pkgs/development/python-modules/guestfs/default.nix:
```diff
@@ -8,12 +9,12 @@
-  version = "1.40.1";
+  version = "1.40.2";
   format = "setuptools";
 
```
```suggestion
  disabled = pythonAtLeast "3.13";
```
And remove meta.broken
